### PR TITLE
feat: add `test` command

### PR DESCRIPTION
**Pull request by pocketken**
This PR adds some limited support for a `test` command:

* `test -e <arg>` - argument exists
* `test -f <arg>` - argument is a regular file (not a directory or symlink)
* `test -d <arg>` - argument is a directory
* `test -s <arg>` - argument is non-zero sized
* `test -L <arg>` - argument is a symlink

@dsherret I could use some guidance on the symlink tests -- I have them ~commented out~ excluded on Windows for the moment as apparently you need Administrative rights to create a symlink:

```
test command => dax/mod.test.ts:180:6
error: Error: A required privilege is not held by the client. (os error 1314), symlink 'zero.dat' -> 'linked.dat'
  await Deno.symlink('zero.dat', 'linked.dat');
  ^
    at async Object.symlink (deno:runtime/js/30_fs.js:367:5)
    at async file:///C:/ ... /dax/mod.test.ts:183:3
```

I am guessing it wouldn't be a problem for GH actions, but I didn't want to assume that was the only place tests would need/want to be run without issues.

I should also note that I've (temporarily) got the `resolvePath` changes that are included in #5 here as I more or less just kept the code the same when I split it off.  My assumption is that would merge first and then I can just update this branch afterward.

### CODE REVIEW

File: mod.test.ts [hunk 1]
```diff
@@ -176,6 +176,99 @@ Deno.test("sleep command", async () => {
   assertEquals(end - start > 190, true);
 });
 
+Deno.test("test command", async (t) => {
+  await Deno.writeFile('zero.dat', new Uint8Array());
+  await Deno.writeFile('non-zero.dat', new Uint8Array([242]));
+  if (Deno.build.os !== 'windows') {
```
Comment (dsherret) [inline mod.test.ts:182]: Do you have developer mode enabled on your machine? https://blogs.windows.com/windowsdeveloper/2016/12/02/symlinks-windows-10/ (I use Windows 10 btw)

That said, I think excluding this test on Windows is fine.

**Review by dsherret — APPROVED** [review]
LGTM. Thanks a lot!

File: mod.test.ts (hunk 1 above)
Comment (pocketken) [reply mod.test.ts:182]: Looks like its failing in GH actions anyway.  I don't technically need the symlink stuff for my current use case, but, it seems like this should work?

Comment (dsherret) [reply mod.test.ts:182]: I feel like it should work too. I'm just looking into it.

Comment (dsherret) [reply mod.test.ts:182]: Oh, Deno.stat always follows symlinks. I'm just looking into an alternative.

**Pull request merged**